File: CSharp/GenericTree/TreeNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GenericTree
{
    class TreeNode<T> where T : IComparable
    {
        public T data { get; set; }

        public TreeNode<T> leftNode { get; set; }

        public TreeNode<T> rightNode { get; set; }

        public TreeNode(T myData)
        {
            data = myData;
        }

        public void Insert(T newData)
        {
            if (newData.CompareTo(data) < 0)
            {
                if (leftNode == null)
                {
                    leftNode = new TreeNode<T>(newData);
                }
                else
                {
                    leftNode.Insert(newData);
                }
            }else if(newData.CompareTo(data) > 0)
            {
                if(rightNode == null)
                {
                    rightNode = new TreeNode<T>(newData);
                }else
                {
                    rightNode.Insert(newData);
                }
            }
        }
    }
}

File: Java/src/IntegerTree/TreeNode.java
package IntegerTree;

public class TreeNode{
	
	public int data;

    public TreeNode leftNode;

    public TreeNode rightNode;

    public TreeNode(int myData)
    {
        data = myData;
    }

    public void Insert(int newData)
    {
        if (newData < this.data)
        {
            if (leftNode == null)
            {
                leftNode = new TreeNode(newData);
            }
            else
            {
                leftNode.Insert(newData);
            }
        }else if(newData > this.data)
        {
            if(rightNode == null)
            {
                rightNode = new TreeNode(newData);
            }else
            {
                rightNode.Insert(newData);
            }
        }
    }
    

	

}
